ndk/hooks/useUser.tsx

```diff
@@ -1,16 +1,23 @@
 import { useEffect, useState } from "react";
-import { NDKUser } from "@nostr-dev-kit/ndk";
-import { useUsers } from "./useUsers";
+import { getCachedUser } from "../inMemoryCacheAdapter";
+import { useNDK } from "../NDKProvider";
 
-export function useUser(pubkey?: string): NDKUser | undefined {
-  const users = useUsers(pubkey ? [pubkey] : []);
-  const [user, setUser] = useState<NDKUser | undefined>();
+export function useUser(pubkey?: string) {
+  const { ndk } = useNDK();
+  const [user, setUser] = useState(getCachedUser(pubkey, ndk));
```

Comment:
with the preloading initial render of all notes in the viewport should have user data